Gemfile

```diff
@@ -14,17 +14,20 @@ gem 'jbuilder', '~> 2.5'
 gem 'jquery-rails'
 gem 'kaminari'
 gem 'mini_magick'
+gem 'pdfkit'
 gem 'pg', '~> 0.18'
 gem 'puma', '~> 3.7'
 gem 'rails', '~> 5.1.4'
 gem 'rails_best_practices'
 gem 'reek', '~> 4.7', '>= 4.7.3'
+gem 'render_anywhere'
```

Comment:
remove this, this looks horrible

Reply:
https://medium.com/evil-martians/new-feature-in-rails-5-render-views-outside-of-actions-2fc1181e86a8
or keep it.
Since Rails 5+ allows to call render from any place, it's not a problem to use this solution if it simplifies life.
The more important problem that report is stored in public folder with the same name.

Reply:
it's a lot easier to work with standalone class for PDF generation (ak PDF Generator, or PDF wrapper) that returns PDF IO and call `#render` from an action,

Reply:
Developer - add this gem
https://github.com/mileszs/wicked_pdf
follow this tutorial - http://prntscr.com/hu3yv1